pkg/plugins/golang/v4/api.go

```diff
@@ -0,0 +1,191 @@
+	defaultCRDVersion = "v1"
+)
+
+// DefaultMainPath is default file path of main.go
+const DefaultMainPath = "main.go"
+
+var _ plugin.CreateAPISubcommand = &createAPISubcommand{}
+
+type createAPISubcommand struct {
+	config config.Config
+
+	options *goPlugin.Options
+
+	resource *resource.Resource
+
+	// Check if we have to scaffold resource and/or controller
+	resourceFlag   *pflag.Flag
+	controllerFlag *pflag.Flag
+
+	// force indicates that the resource should be created even if it already exists
+	force bool
+
+	// runMake indicates whether to run make or not after scaffolding APIs
+	runMake bool
+}
+
+func (p *createAPISubcommand) UpdateMetadata(cliMeta plugin.CLIMetadata, subcmdMeta *plugin.SubcommandMetadata) {
+	subcmdMeta.Description = `Scaffold a Kubernetes API by writing a Resource definition and/or a Controller.
+
+If information about whether the resource and controller should be scaffolded
+was not explicitly provided, it will prompt the user if they should be.
+
+After the scaffold is written, the dependencies will be updated and
+make generate will be run.
+`
+	subcmdMeta.Examples = fmt.Sprintf(`  # Create a frigates API with Group: ship, Version: v1beta1 and Kind: Frigate
+  %[1]s create api --group ship --version v1beta1 --kind Frigate
+
+  # Edit the API Scheme
+  nano api/v1beta1/frigate_types.go
+
+  # Edit the Controller
+  nano controllers/frigate/frigate_controller.go
+
+  # Edit the Controller Test
+  nano controllers/frigate/frigate_controller_test.go
+
+  # Generate the manifests
+  make manifests
+
+  # Install CRDs into the Kubernetes cluster using kubectl apply
+  make install
+
+  # Regenerate code and run against the Kubernetes cluster configured by ~/.kube/config
+  make run
+`, cliMeta.CommandName)
+}
+
+func (p *createAPISubcommand) BindFlags(fs *pflag.FlagSet) {
+	fs.BoolVar(&p.runMake, "make", true, "if true, run `make generate` after generating files")
+
+	fs.BoolVar(&p.force, "force", false,
+		"attempt to create resource even if it already exists")
+
+	p.options = &goPlugin.Options{}
+
+	fs.StringVar(&p.options.Plural, "plural", "", "resource irregular plural form")
+
+	fs.BoolVar(&p.options.DoAPI, "resource", true,
+		"if set, generate the resource without prompting the user")
+	p.resourceFlag = fs.Lookup("resource")
+	fs.BoolVar(&p.options.Namespaced, "namespaced", true, "resource is namespaced")
+
+	fs.BoolVar(&p.options.DoController, "controller", true,
+		"if set, generate the controller without prompting the user")
+	p.controllerFlag = fs.Lookup("controller")
+}
+
+func (p *createAPISubcommand) InjectConfig(c config.Config) error {
+	p.config = c
+	// go/v4 no longer supports v1beta1 option
+	p.options.CRDVersion = defaultCRDVersion
+	return nil
+}
+
+func (p *createAPISubcommand) InjectResource(res *resource.Resource) error {
+	p.resource = res
+
+	// TODO: re-evaluate whether y/n input still makes sense. We should probably always
+	//       scaffold the resource and controller.
```

Comment:
Agree to directly generate resources and controller code with asking users.
Assuming these are the common patterns to implement CRD operators.

Reply:
Hun, we need to remove this TODO.
It cames from the go v3 code as well.
We need to have the Y/N option. WHY?

We can reconcile a Kind that is not owned by the project.
That can be a core type (i.e Deployments) or an external/third-party one like a CRD created by other projects. See: https://github.com/kubernetes-sigs/kubebuilder/issues/3082

Lets remove it in a follow up.

Reply:
Sounds good. Makes sense to leave options for users considering on such wider use cases.